core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/AwarenessAllocationDecider.java

```diff
@@ -234,7 +234,9 @@ private Decision underCapacity(ShardRouting shardRouting, RoutingNode node, Rout
             int currentNodeCount = shardPerAttribute.get(node.node().attributes().get(awarenessAttribute));
             // if we are above with leftover, then we know we are not good, even with mod
             if (currentNodeCount > (requiredCountPerAttribute + leftoverPerAttribute)) {
-                return allocation.decision(Decision.NO, NAME, "too many shards on nodes for attribute: [%s]", awarenessAttribute);
+                return allocation.decision(Decision.NO, NAME,
+                        "too many shards on node for attribute: [%s], required per attribute: [%s], node count: [%s], leftover: [%s]",
+                        awarenessAttribute, requiredCountPerAttribute, currentNodeCount, leftoverPerAttribute);
```

Comment:
I am just curious why not `"too many shards on node for attribute: [%s], required per attribute: [%d], node count: [%d], leftover: [%d]"`?